v1/api/publisher.ts

```diff
@@ -36,9 +39,47 @@ export const publisherRoutes = (store: StoreI) => async (server: FastifyInstance
       description: 'Gets information about a specific publisher',
       tags: ['publisher']
     }
-  }, async (request, _reply) => {
+  }, async (request, reply) => {
     const { id } = request.params
-    return store.publisher.get(id)
+    return reply.send(await store.publisher.get(id))
+  })
+
+  server.get<{
+    Params: {
+      id: string
+    }
+    Reply: string
+  }>('/publisher/:id/auth', {
+    schema: {
+      params: {
+        id: Type.String()
+      },
+      description: 'Retrieves refresh token for specified publisher. Admin gated',
```

Comment:
Hmm, I'm rethinking some of this token stuff.

It might be good to think about "capabilities" for tokens right from the get go.

E.g., if a token is and `admin` token, it can do anything.

If it's a publisher token, it can have the "refresh" capability by default, and this can be used to generate a token with the same capabilities but with a later expiration time.

If it doesn't have the refresh capability, the new token should have the same expiration.

From here we can get rid of the `/refresh` endpoint entirely and just focus on the token creation endpoint.

It's also how we can scale to creating scoped tokens for specific domains and the such. 

Does it make sense to get to this sooner than later?

Maybe we can also add a URL for the root of who this is for, e.g. if it's a publisher token we can have a URL pointing to `https://domain.com/publisher/:id/`

Reply:
So the workflow for a client would be:

1. Decide what I want to do
2. I'm gonna update the website
3. Use my long lived ("refresh") token to issue an update-website token for my site sutty.ml
4. Stream a tarball for sutty.ml with the update-website token attached
5. Forget the update-website token
6. Finish

Something like that?

Reply:
Hey, just seeing this now. @fauno, I think @RangerMauve means that a token with a 'publisher' capability can be used to exchange for a new publisher token with a later expiration time

This makes sense to me, I can draft up the changes